lib/client.js

```diff
@@ -21,6 +21,10 @@ var defaultConnectOptions = {
   protocolId: 'MQTT',
   protocolVersion: 4,
   reconnectPeriod: 1000,
+  backOffStrategy: null,
```

Comment:
I don't know, how I could make the backoff strategy more modular. It relies on some member variables and I could'nt think of a generalization of clearInterval/clearTimeout along other algorithm specific variables :-/
Could you explain your thoughts on this a little?